typescript/web/src/components/invitation-manager/invitation-container.tsx

```diff
@@ -0,0 +1,44 @@
+import {
+  Center,
+  Flex,
+  Heading,
+  useColorModeValue as mode,
+  VStack,
+} from "@chakra-ui/react";
+import { PropsWithChildren } from "react";
+
+type InvitationContainerProps = {
+  header: string;
+  details: string;
+};
+
+type CompleteProps = PropsWithChildren<InvitationContainerProps>;
+
+export const InvitationContainer = ({
+  header,
+  details,
+  children,
+}: CompleteProps) => {
+  return (
+    <Center h="full">
+      <VStack
+        p={6}
+        spacing={6}
+        align="start"
+        bg={mode("white", "gray.800")}
+        borderRadius="md"
+        boxShadow="lg"
+        maxW="2xl"
+        my="3.75rem"
+      >
+        <Heading fontSize="xl" fontWeight="semibold">
+          {header}
+        </Heading>
+        <Flex>{details}</Flex>
+        <Flex w="100%" justify="flex-end">
```

Comment:
There must be a cleaner way than using `w="100%"` no?

Reply:
Apart from the fact that is ignores the padding and some other props, the parent is a flex container so using `width="100%"` is kind of an heresy. You should be able use `alignSelf="stretch"`  or something like that for such cases.

Reply:
Not in all cases. If the text is small enough, the VStack will shrink
I've used it anyway and put a `w="full"` above it